internal/tasks/fleet_rollout.go

```diff
@@ -194,6 +200,46 @@ func (f FleetRolloutsLogic) updateDeviceToFleetTemplate(ctx context.Context, dev
 	return err
 }
 
+func (f FleetRolloutsLogic) getDeviceApps(device *api.Device, templateVersion *api.TemplateVersion) (*[]api.ApplicationSpec, error) {
+	if templateVersion.Status.Applications == nil {
+		return nil, nil
+	}
+
+	deviceApps := []api.ApplicationSpec{}
+	for _, app := range *templateVersion.Status.Applications {
+		appType, err := app.Type()
```

Comment:
Something is off here: The service cannot know anything about the application type, because for that it'd need to read the application image's labels, right? And it may not have access to the OCI registry.

Reply:
yeah its actually the provider type I can make that more clear.